source/agora/consensus/data/Enrollment.d

```diff
@@ -70,6 +70,21 @@ public struct Enrollment
     }
 }
 
+/*******************************************************************************
+
+    Define enrollment stored in the pool
+
+*******************************************************************************/
+
+public struct PoolEnrollment
+{
+    /// An enrollment data in the pool
+    public Enrollment enrollment;
+
+    /// The available height set on being stored in the pool
+    public Height avail_height;
+}
```

Comment:
I don't follow why we need to get the `avail_height` from other nodes ?

Reply:
The `avail_height` is needed to store enrollments into the pool because a requesting node could have a different height and can not set the exact value for the field.

Reply:
Anyway, I'll look into more whether the field will be removed.

Reply:
I think it's needed. So I remain the field as it is.

Reply:
The reason I'm asking is because we cannot trust the remote node to provide the right information.
What if the node sends us a right enrollment but pretends it will be available in 3 million blocks ?

Reply:
After considering your opinion. I could use a new table for missing enrollments that have `Enrollment` and `avail_height`. How do you think of creating new table for that?

Reply:
Anyway, I will find a way to use `Enrollment` in the API endpoint, `getEnrollments`, and remove this structure.